src/frontend/src/components/LinkEditView.tsx

```diff
@@ -0,0 +1,54 @@
+import { useAllLinkTypes } from '../hooks/projects.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import ErrorPage from '../pages/ErrorPage';
+import { Button, Grid, IconButton, MenuItem, Select, TextField } from '@mui/material';
+import { FieldArrayWithId, UseFieldArrayAppend, UseFieldArrayRemove, UseFormRegister } from 'react-hook-form';
+import DeleteIcon from '@mui/icons-material/Delete';
+
+const LinkEditView: React.FC<{
+  name: string;
+  ls: FieldArrayWithId[];
+  register: UseFormRegister<any>;
+  append: UseFieldArrayAppend<any, any>;
+  remove: UseFieldArrayRemove;
+}> = ({ name, ls, register, append, remove }) => {
+  const { isLoading, isError, error, data } = useAllLinkTypes();
+  if (isLoading || !data) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  return (
+    <>
+      {ls.map((_element, i) => {
```

Comment:
not very pretty

Reply:
I copied and pasted this from the react hook edit list component we have, we should prob refactor but im not very familiar with use field array yet